aptos-move/move-examples/mint_nft_v2/2-Using-Resource-Account/sources/create_nft_with_resource_account.move

```diff
@@ -0,0 +1,201 @@
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_signer, @source_addr);
+        let resource_address = signer::address_of(resource_signer);
+        let collection_object_address = collection::create_collection_address(&resource_address, &collection_name);
+
+        // Store the token data id and the resource account's signer capability within the module, so we can programmatically
+        // sign for transactions in the `mint_event_ticket()` function.
+        move_to(resource_signer, ModuleData {
+            signer_cap: resource_signer_cap,
+            collection_name,
+            token_description: string::utf8(b""),
+            token_name,
+            token_uri,
+            property_keys: vector<String>[string::utf8(b"given_to")],
+            property_types: vector<String>[ string::utf8(b"address") ],
+            property_values: vector<vector<u8>>[bcs::to_bytes(&@source_addr)],
+            collection_object_address,
+        });
+    }
+
+    /// Mint an NFT to the receiver. Note the difference from the tutorial in part 1, here we only ask for the receiver's
+    /// address. This is because we used resource account to publish this module and stored the resource account's signer
+    /// within the `ModuleData`, so we can programmatically sign for transactions instead of manually signing transactions.
+    /// See https://aptos.dev/concepts/accounts/#resource-accounts for more information about resource account.
+    /// Note also the difference with Token v2: we no longer need the receiver to sign the transaction. You can send objects
+    /// to accounts without the prior approval of the account you're sending objects to.
+    public entry fun mint_event_ticket(receiver: &signer) acquires ModuleData {
+        let module_data = borrow_global_mut<ModuleData>(@mint_nft_v2);
+
+        // Create a signer of the resource account from the signer capabiity stored in this module.
+        // Using a resource account and storing its signer capability within the module allows the module to programmatically
+        // sign transactions on behalf of the module.
+        let resource_signer = account::create_signer_with_capability(&module_data.signer_cap);
+        let resource_address = signer::address_of(&resource_signer);
+        let token_creation_num = account::get_guid_next_creation_num(resource_address);
+
+        aptos_token::mint(
+            &resource_signer,
+            module_data.collection_name,
+            module_data.token_description,
+            module_data.token_name,
+            module_data.token_uri,
+            module_data.property_keys,
+            module_data.property_types,
+            module_data.property_values,
+        );
+
+        let token_object = object::address_to_object<AptosToken>(object::create_guid_object_address(resource_address, token_creation_num));
```

Comment:
Hmm this adds more blockers for parallelism (beside the supply being tracked) as it uses a single account's guid. Let's add a quick comment to mention this and we can come back and fix this later once we have the pieces ready for full parallelism